pkg/kp/constants.go

```diff
@@ -1,38 +1,147 @@
 package kp
 
 import (
-	"strings"
+	"path"
+
+	rc_fields "github.com/square/p2/pkg/rc/fields"
+	"github.com/square/p2/pkg/types"
+	"github.com/square/p2/pkg/util"
 )
 
+// PodPrefix is a subset of TreeType. A valid PodPrefix is a TreeType that has
+// pods under it, for example "intent"
+type PodPrefix string
+
+// A TreeType represents a top level tree root in the kv store, e.g. "intent"
+// or "lock"
+type TreeType string
+
 const (
-	INTENT_TREE  string = "intent"
-	REALITY_TREE string = "reality"
-	HOOK_TREE    string = "hooks"
-	LOCK_TREE    string = "lock"
-	RC_TREE      string = "replication_controllers"
-	ROLL_TREE    string = "rolls"
+	INTENT_TREE  PodPrefix = "intent"
+	REALITY_TREE PodPrefix = "reality"
+	HOOK_TREE    PodPrefix = "hooks"
+	LOCK_TREE    TreeType  = "lock"
+	RC_TREE      TreeType  = "replication_controllers"
+	ROLL_TREE    TreeType  = "rolls"
+)
+
+var (
+	VALID_POD_PREFIXES = []PodPrefix{
```

Comment:
It seems awkward to have to list each prefix three times to establish its full type. I suspect you could make this a lot cleaner if you expressed the concepts of "this is a tree prefix" and "this is a tree prefix that contains pods" using interfaces.